agent/pagebroker/daemon.cpp

```
@@ -0,0 +1,147 @@
+  std::strcpy(address.sun_path, socket_path.c_str());
+  if (bind(listener.get(), reinterpret_cast<const sockaddr*>(&address), sizeof(address)) < 0 ||
+      listen(listener.get(), 16) < 0) {
+    std::cerr << "listen: " << std::strerror(errno) << '\n';
+    return 1;
+  }
+
+  Broker broker(argv[kStagingDirectory]);
+  for (;;) {
+    FileDescriptor connection(accept(listener.get(), nullptr, nullptr));
+    if (connection.get() < 0) {
+      if (errno != EINTR)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)